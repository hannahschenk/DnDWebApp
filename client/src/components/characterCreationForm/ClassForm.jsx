import React, {useState, useEffect} from 'react';
import dndApi from "../../utils/dnd5eApi";

const ClassForm = () => {
    const [classChoices, setClassChoices]  = useState([]);

    useEffect(() => {
        dndApi.getClasses()
        .then((response) => setClassChoices(response.data.results))
        .catch(err => console.log(err))
    }, [])


    const pickClass = (chosenClassInfo) => {
        dndApi.getMoreInfo(chosenClassInfo.url)
        .then(data => {
            console.log(data.response)
            /*
                TODO: this is a good spot to format the data and send what ever we need to the global state
            */
        })
        .catch(err => console.log(err))
    }

    //note: having a form here is kind of useless but for the sake of being semantic
    return (
        <>
            <form>
                <p>Pick a Class: </p>
                {
                    classChoices.map((classContent) => 
                        <React.Fragment key={`${classContent.index}`}>
                            <label htmlFor="class">{classContent.name}</label>
                            <input 
                                type="radio" 
                                name="class"
                                id="class"
                                value={classContent.index}
                                onClick={() => pickClass(classContent)}
                            /><br/>
                        </React.Fragment>
                    )
                }
            </form>
        </>
    );
};

export default ClassForm;