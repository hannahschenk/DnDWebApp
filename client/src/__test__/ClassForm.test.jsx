import React, { useState as useStateMock } from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import ClassForm from "../components/characterCreationForm/ClassForm";

describe("The AbilityScoresForm Component...", () => {
    test("will render without crashing", () => {
        const { baseElement } = render(<ClassForm />);
        expect(baseElement).toBeDefined();
    })
 
    test("will match snapshot",()=>{
        // renderer will give you a pure JS object you can turn in to JSON of the page the has rendered.
        const tree = renderer.create(<ClassForm />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})