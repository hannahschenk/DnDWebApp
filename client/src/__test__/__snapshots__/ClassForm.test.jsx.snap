// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AbilityScoresForm Component... will match snapshot 1`] = `
<form>
  <p>
    Pick a Class: 
  </p>
</form>
`;
